### ISSUE TITLE
Reimplement torch conv ops to go direct to cudnn

### ISSUE THREAD

**Issue by fchollet** [keras-team/keras#18457]
Right now our conv ops in torch are a bit slow, because torch has non-standard semantics which force us to convert back and forth:

- NCHW layout
- different kernel layout
- different padding conventions

We should reimplement them in a way that goes direct to cudnn on GPU. Not only would it fix the performance regression, it's likely to end faster than vanilla PyTorch.
